oshi-core/src/main/java/oshi/util/platform/windows/RegistryUtil.java

```diff
@@ -0,0 +1,181 @@
+import com.sun.jna.platform.win32.Advapi32Util;
+import com.sun.jna.platform.win32.Win32Exception;
+import com.sun.jna.platform.win32.WinReg;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import oshi.util.ParseUtil;
+
+import com.sun.jna.platform.win32.WinReg.HKEY;
+import com.sun.jna.platform.win32.WinReg.HKEYByReference;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Locale;
+
+import static com.sun.jna.platform.win32.WinError.ERROR_SUCCESS;
+import static com.sun.jna.platform.win32.WinNT.KEY_READ;
+
+public final class RegistryUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RegistryUtil.class);
+
+    private static final long THIRTY_YEARS_IN_SECS = 30L * 365 * 24 * 60 * 60;
+
+    private static final Advapi32 ADV = Advapi32.INSTANCE;
+
+    private RegistryUtil() {
+    }
+
+    /**
+     * Returns a registry value as a Long. (without access flag) Currently supports String and Integer
+     */
+    public static long getLongValue(HKEY root, String path, String key) {
+        try {
+            Object val = Advapi32Util.registryGetValue(root, path, key);
+            return registryValueToLong(val);
+        } catch (Win32Exception e) {
+            LOG.trace("Unable to access " + path + ": " + e.getMessage());
+        }
+        return 0L;
+    }
+
+    /**
+     * Returns a registry value as a Long. (with access flag) Currently supports String and Integer
+     */
+    public static long getLongValue(HKEY root, String path, String key, int accessFlag) {
+        Object val = getRegistryValueOrNull(root, path, key, accessFlag);
+        return registryValueToLong(val);
+    }
+
+    public static Object getRegistryValueOrNull(HKEY root, String path, String key, int accessFlag) {
+        HKEY hKey = null;
+        try {
+            hKey = getRegistryHKey(root, path, accessFlag);
+            Object value = Advapi32Util.registryGetValue(root, path, key);
+            if ((value instanceof Integer) || (value instanceof String && !((String) value).trim().isEmpty())
+                    || value instanceof byte[]) {
+                return value;
+            }
+        } catch (Win32Exception e) {
+            LOG.trace("Unable to access " + path + " with flag " + accessFlag + ": " + e.getMessage());
+        } finally {
+            if (hKey != null) {
+                int rc = ADV.RegCloseKey(hKey);
+                if (rc != ERROR_SUCCESS) {
+                    throw new Win32Exception(rc);
+                }
+            }
+        }
+        return null;
+    }
+
+    private static HKEY getRegistryHKey(HKEY rootKey, String path, int accessFlag) {
+        HKEYByReference phkKey = new HKEYByReference();
+        int rc = ADV.RegOpenKeyEx(rootKey, path, 0, KEY_READ | accessFlag, phkKey);
+        if (rc != ERROR_SUCCESS) {
+            throw new Win32Exception(rc);
+        }
+        return phkKey.getValue();
+    }
+
+    private static long registryValueToLong(Object val) {
```

Comment:
Let's add a javadoc (optional for private) or at least comments here indicating the purpose of this method